org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/ServerLauncher.xtend

```diff
@@ -7,61 +7,135 @@
+	def void start(LaunchArgs it) {
+		println("Xtext Language Server is starting.")
+		val launcher = Launcher.createLauncher(languageServer, LanguageClient, in, out, validate, trace)
 		languageServer.connect(launcher.remoteProxy)
 		val future = launcher.startListening
-		System.err.println("started.")
+		println("Xtext Language Server has been started.")
 		while (!future.done) {
 			Thread.sleep(10_000l)
 		}
 	}
 
-	def static redirectStandardStreams() {
-		System.setIn(new ByteArrayInputStream(newByteArrayOfSize(0)))
-		val id = (ServerLauncher.name + "-" + new Timestamp(System.currentTimeMillis)).replace(" ", "_")
-		if (IS_DEBUG) {
-			val stdFileOut = new FileOutputStream("out-" + id + ".log")
-			System.setOut(new PrintStream(stdFileOut))
-			val stdFileErr = new FileOutputStream("error-" + id + ".log")
-			System.setErr(new PrintStream(stdFileErr))
+	def static LaunchArgs createLaunchArgs(String prefix, String[] args) {
+		val launchArgs = new LaunchArgs
+		launchArgs.in = System.in
+		launchArgs.out = System.out
+		redirectStandardStreams(prefix, args)
+		launchArgs.trace = args.trace
+		launchArgs.validate = args.shouldValidate
+		return launchArgs
+	}
+
+	def static PrintWriter getTrace(String[] args) {
+		if (shouldTrace(args))
+			return createTrace
+	}
+
+	def static PrintWriter createTrace() {
+		return new PrintWriter(System.out)
+	}
+
+	def static redirectStandardStreams(String prefix, String[] args) {
+		if (shouldLogStandardStreams(args)) {
+			logStandardStreams(prefix)
 		} else {
-			System.setOut(new PrintStream(new ByteArrayOutputStream()))
-			System.setErr(new PrintStream(new ByteArrayOutputStream()))
+			silentStandardStreams
 		}
 	}
 
+	def static boolean shouldValidate(String[] args) {
+		return !args.testArg(NO_VALIDATE)
+	}
+
+	def static boolean shouldTrace(String[] args) {
+		return args.testArg(TRACE)
+	}
+
+	def static boolean shouldLogStandardStreams(String[] args) {
+		return args.testArg(ServerLauncher.LOG, 'debug')
+	}
+
+	def static boolean testArg(String[] args, String ... values) {
+		return args.exists[arg|arg.testArg(values)]
+	}
+
+	def static boolean testArg(String arg, String ... values) {
+		return values.exists[value|value == arg]
+	}
+
+	def static void logStandardStreams(String prefix) {
+		val stdFileOut = new FileOutputStream(prefix + "-debug.log")
+		val stdFileErr = new FileOutputStream(prefix + "-error.log")
+		redirectStandardStreams(stdFileOut, stdFileErr)
+	}
+
+	def static void silentStandardStreams() {
+		redirectStandardStreams(ServerLauncher.silentOut, ServerLauncher.silentOut)
+	}
+
+	def static void redirectStandardStreams(OutputStream out, OutputStream err) {
+		redirectStandardStreams(ServerLauncher.silentIn, out, err)
+	}
+
+	def static void redirectStandardStreams(InputStream in, OutputStream out, OutputStream err) {
+		System.setIn(in)
+		System.setOut(new PrintStream(out))
+		System.setErr(new PrintStream(err))
```

Comment:
I see that uncaught exceptions end up in `System.err`. If we don't redirect it, I think, it will break communication.